src/test/java/com/amazon/ion/impl/IonRawTextReaderTest_1_1.java

```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    static Parameter[] validParameters() {
+        return new Parameter[] {
+            Parameter.of("$ion_1_1 (:foo)", "foo", null, false, ExpressionType.E_EXPRESSION),
+            Parameter.of("$ion_1_1 (:foo bar)", "foo", "bar", false, ExpressionType.E_EXPRESSION),
+            Parameter.of("$ion_1_1 (::foo)", "foo", null, false, ExpressionType.EXPRESSION_GROUP), // TODO do we want to require whitespace after ::?
+            Parameter.of("$ion_1_1 (:: foo bar)", "foo", "bar", false, ExpressionType.EXPRESSION_GROUP),
+            Parameter.of("$ion_1_1 (.foo)", ".", "foo", false, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (.. foo)", "..", "foo", false, ExpressionType.NONE),
+            Parameter.of("$ion_1_0 (.foo)", ".", "foo", false, ExpressionType.NONE),
+            Parameter.of("$ion_1_0 (.. foo)", "..", "foo", false, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (.foo)", "foo", null, true, ExpressionType.E_EXPRESSION),
+            Parameter.of("$ion_1_1 (.. foo)", "foo", null, true, ExpressionType.EXPRESSION_GROUP),
+            Parameter.of("$ion_1_1 (.+ foo)", ".+", "foo", true, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (..+ foo)", "..+", "foo", true, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (.+ foo)", ".+", "foo", false, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (..+ foo)", "..+", "foo", false, ExpressionType.NONE),
+        };
+    }
+
+    @ParameterizedTest
+    @MethodSource("validParameters")
+    public void validExpressionSyntax(Parameter parameter) throws Exception {
+        try (IonReader reader = parameter.newTextReader()) {
+            reader.next();
+            reader.stepIn();
+            assertEquals(IonType.SYMBOL, reader.next());
+            assertEquals(parameter.firstSymbol, reader.stringValue());
+            parameter.expressionType.verifyExpressionType(reader);
+            if (parameter.secondSymbol == null) {
+                assertNull(reader.next());
+            }
+            if (parameter.secondSymbol != null) {
+                assertEquals(IonType.SYMBOL, reader.next());
+                assertEquals(parameter.secondSymbol, reader.stringValue());
+                assertNull(reader.next());
+            }
+            reader.stepOut();
+            assertNull(reader.next());
+        }
+    }
+
+    static Parameter[] invalidParameters() {
+        return new Parameter[] {
+            // Colon is not a valid operator in Ion 1.0.
+            Parameter.of("$ion_1_0 (:foo)", null, null, false, ExpressionType.NONE),
+            Parameter.of("$ion_1_0 (::foo)", null, null, false, ExpressionType.NONE),
+            // Colon is not a valid operator in Ion 1.1 TDL.
+            Parameter.of("$ion_1_1 (:foo)", null, null, true, ExpressionType.NONE),
+            Parameter.of("$ion_1_1 (::foo)", null, null, true, ExpressionType.NONE),
```

Comment:
...but this is not TDL here. Can you add some comments or slightly more descriptive test names?

Reply:
The test forces the raw reader to consider the content to be TDL (via the boolean `true` in the second to last argument). I'll add some comments to make that more clear.